client/app/scripts/superdesk-ingest/module.js

```diff
@@ -1031,11 +1031,85 @@ define([
         };
     }
 
-    IngestRoutingSchedule.$inject = [];
-    function IngestRoutingSchedule() {
+    /**
+     * @memberof superdesk.ingest
+     * @ngdoc directive
+     * @name sdIngestRoutingSchedule
+     * @description
+     *   Creates the Schedule section (tab) of the routing rule edit form.
+     */
+    IngestRoutingSchedule.$inject = ['tzdata'];
+    function IngestRoutingSchedule(tzdata) {
         return {
-            scope: {rule: '='},
-            templateUrl: 'scripts/superdesk-ingest/views/settings/ingest-routing-schedule.html'
+            scope: {
+                rule: '='  // the routing rule whose schedule is being edited
+            },
+            templateUrl: 'scripts/superdesk-ingest/views/settings/ingest-routing-schedule.html',
+            link: function (scope, element, attrs) {
+
+                scope.timeZones = [];     // all time zones to choose from
+                scope.tzSearchTerm = '';  // the current time zone search term
+
+                // filtered time zone list containing only those that match
+                // user-provided search term
+                scope.matchingTimeZones = [];
+
+                tzdata.$promise.then(function () {
```

Comment:
What do you mean, the `$promise` part or the `_.union()` logic?
The former is needed because TZ data is fetched asynchronously and services that use it must know when the data has arrived.
The latter could be done, the "unioned" data could be pre-caclulated and stored as ... hm, some sort of a property of the `tzdata` Array? Is that what you had in mind?